### PR TITLE
Add JS target

### PR DESCRIPTION
**Pull request by lukaville**

### CODE REVIEW

File: js-browser-sample-app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,35 @@
+plugins {
```
Comment (arkivanov) [inline js-browser-sample-app/build.gradle]: Let's rename name our folders wit samples in the following way: sample-xxx (both android and this one)

File: js-browser-sample-app/src/main/kotlin/com/badoo/main.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                src = kitten.url
+            })
+        }
+}
+
+class Kitten(val url: String)
+
+fun loadRandomKitten(): Single<List<Kitten>> {
+    return window
+        .fetch("https://api.thecatapi.com/v1/images/search")
```
Comment (arkivanov) [inline js-browser-sample-app/src/main/kotlin/com/badoo/main.kt]: singleFromFunction?

File: js-browser-sample-app/src/main/kotlin/com/badoo/main.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        .toSingle()
+        .flatMap { it.json().toSingle() }
```
Comment (arkivanov) [inline js-browser-sample-app/src/main/kotlin/com/badoo/main.kt]: Why flatMap, maybe just map? And method reference.

File: js-browser-sample-app/src/main/kotlin/com/badoo/main.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        .map {
+            (it.asDynamic() as Array<dynamic>).map { kitten ->
+                Kitten(kitten.url as String)
+            }
+        }
+}
+
+fun <T> Promise<T>.toSingle(): Single<T> = singleByEmitter { emitter ->
```
Comment (arkivanov) [inline js-browser-sample-app/src/main/kotlin/com/badoo/main.kt]: Should it be a part of library (jsMain source set)?

File: js-browser-sample-app/src/main/kotlin/com/badoo/main.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    then(
+        onFulfilled = { emitter.onSuccess(it) },
```
Comment (arkivanov) [inline js-browser-sample-app/src/main/kotlin/com/badoo/main.kt]: Better to use method references

File: reaktive/src/jsMain/kotlin/com/badoo/reaktive/scheduler/CreateIoScheduler.kt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package com.badoo.reaktive.scheduler
+
+actual fun createIoScheduler(): Scheduler =
+    throw NotImplementedError("IO scheduler is not supported in JS target")
```
Comment (arkivanov) [inline reaktive/src/jsMain/kotlin/com/badoo/reaktive/scheduler/CreateIoScheduler.kt]: Perhaps it's better to fallback to main scheduler?

File: reaktive/src/jsMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package com.badoo.reaktive.scheduler
+
+import com.badoo.reaktive.disposable.CompositeDisposable
+import kotlin.browser.window
+
+class MainScheduler : Scheduler {
```
Comment (arkivanov) [inline reaktive/src/jsMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt]: internal

File: reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/UncaughtErrorHandler.kt [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package com.badoo.reaktive.utils
+
+actual var reaktiveUncaughtErrorHandler: (Throwable) -> Unit =
+    { e -> console.error(e) }
```
Comment (arkivanov) [inline reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/UncaughtErrorHandler.kt]: Method reference

File: js-browser-sample-app/src/main/kotlin/com/badoo/main.kt (hunk 4 above)
Comment (mk-jones) [reply js-browser-sample-app/src/main/kotlin/com/badoo/main.kt]: I agree that it should be part of the library.

File: reaktive/src/jsMain/kotlin/com/badoo/reaktive/scheduler/CreateIoScheduler.kt (hunk 6 above)
Comment (mk-jones) [reply reaktive/src/jsMain/kotlin/com/badoo/reaktive/scheduler/CreateIoScheduler.kt]: Also agree with this. Will make using the library easier.

File: js-browser-sample-app/src/main/kotlin/com/badoo/main.kt (hunk 2 above)
Comment (lukaville) [reply js-browser-sample-app/src/main/kotlin/com/badoo/main.kt]: it doesn't handle errors

File: js-browser-sample-app/src/main/kotlin/com/badoo/main.kt (hunk 3 above)
Comment (lukaville) [reply js-browser-sample-app/src/main/kotlin/com/badoo/main.kt]: json() method is asynchronous

File: reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/UncaughtErrorHandler.kt (hunk 8 above)
Comment (lukaville) [reply reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/UncaughtErrorHandler.kt]: `console.error(vararg o: Any?): Unit` cannot be used as a method reference for `(Throwable) -> Unit`

File: reaktive/src/jsMain/kotlin/com/badoo/reaktive/promise/PromiseToReaktive.kt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.badoo.reaktive.promise
+
+import com.badoo.reaktive.single.Single
+import com.badoo.reaktive.single.singleByEmitter
+import kotlin.js.Promise
+
+fun <T> Promise<T>.toReaktive(): Single<T> = singleByEmitter { emitter ->
```
Comment (arkivanov) [inline reaktive/src/jsMain/kotlin/com/badoo/reaktive/promise/PromiseToReaktive.kt]: Let's follow our naming style: asSingle

Comment (arkivanov) [reply reaktive/src/jsMain/kotlin/com/badoo/reaktive/promise/PromiseToReaktive.kt]: Don't we need cancellation?

Comment (lukaville) [reply reaktive/src/jsMain/kotlin/com/badoo/reaktive/promise/PromiseToReaktive.kt]: It's not possible to cancel a promise

**Pull request merged**